lib/Runtime/Language/InterpreterStackFrame.cpp

```diff
@@ -1543,29 +1536,30 @@ namespace Js
                 push eax; // push the return value into the stack
                 push layout;
                 call InterpreterStackFrame::GetDynamicRetType;
-                cmp eax, Void;
-                je end;
-                cmp eax, Signed;
-                je end;
-                cmp eax, Float;
-                jne skipFloat;
-                // float
+                mov ecx, eax;
```

Comment:
http://x86.renejeschke.de/html/file_module_x86_id_285.html
`shl ecx, al` is not a legal operation. only `cl` is legal to do a shift using a register